backend/apps/sandbox/views.py

```
@@ -468,3 +468,37 @@ class MaintainerEvaluationViewSet(viewsets.ModelViewSet):
+
+from .models import PipelineExecution, PipelineJob
+from .serializers import PipelineExecutionSerializer
+
+
+class PipelineExecutionViewSet(viewsets.ModelViewSet):
+    """
+    ViewSet for the CI/CD Pipeline Simulator.
+    Creating a pipeline will immediately run the simulation and populate all job logs.
+    """
+    serializer_class = PipelineExecutionSerializer
+    permission_classes = [permissions.IsAuthenticated]
```

Ruff (0.15.21)

[warning] Mutable default value for class attribute

(RUF012)